rust/sdk-examples/src/utils/database.rs

```diff
@@ -179,16 +179,19 @@ where
     debug!("Executing query: {:?}", debug_string);
     let conn = &mut pool.get().await.map_err(|e| {
         warn!("Error getting connection from pool: {:?}", e);
-        diesel::result::Error::DatabaseError(
-            diesel::result::DatabaseErrorKind::UnableToSendCommand,
-            Box::new(e.to_string()),
-        )
+        ProcessorError::DBStoreError {
+            message: e.to_string(),
+            query: Some(debug_string.clone()),
+        }
     })?;
     let res = final_query.execute(conn).await;
     if let Err(ref e) = res {
         warn!("Error running query: {:?}\n{:?}", e, debug_string);
     }
-    res
+    res.map_err(|e| ProcessorError::DBStoreError {
+        message: e.to_string(),
```

Comment:
You might get more context if you do this:
```suggestion
        message: format!("{:#}", e),
```